### ISSUE TITLE
Disable sign on pull requests

### ISSUE THREAD

**Issue by rmadsen-ks** [opentap/opentap#585]
Only do sign on main and release branches.

This may help us get rid of this:

![image](https://user-images.githubusercontent.com/24250062/168805697-fc81087a-cd28-4b86-bfdd-6167b0602794.png)

**Comment by g-galante** [comment]
I think the message is produced when the `CI` jobs _living_ in `environment: sign`: 
- `Package-Win`
- `Package-Linux`
- `Installer-Windows`
- `Docker-Linux`

run.

Actually, it is not clear to me why you use an `environment` for the first three jobs that don't perform any deployment. Maybe it's just for encapsulating some secrets, but it's confusing.

The only job that seems to deserve an `environment` is `Docker-Linux`, and it would make more sense to name it `docker.com`.

**Comment by rmadsen-ks** [comment]
You are right for the first three, we use environment as a way of encapsulating secrets.  Maybe this is not the best approach. Do know a better solution?
